vermouth/processors/apply_rubber_band.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2018 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+import numpy as np
+import networkx as nx
+from .processor import Processor
+from .. import selectors
+
+
+def self_distance_matrix(coordinates):
+    return np.sqrt(((coordinates[:, np.newaxis, :] - coordinates[np.newaxis, :, :]) ** 2).sum(axis=-1))
+
+
+def compute_decay(distance_matrix, shift, rate, power):
+    return np.exp(-rate * ((distance_matrix - shift) **  power))
+
+
+def compute_force_constants(distance_matrix, lower_bound, upper_bound,
+                            decay_factor, decay_power, base_constant,
+                            minimum_force):
+    constants = compute_decay(distance_matrix, lower_bound, decay_factor, decay_power)
+    np.fill_diagonal(constants, 0)
+    constants *= base_constant
+    constants[constants < minimum_force] = 0
+    constants[distance_matrix > upper_bound] = 0
```

Comment:
`constants[(constants < minimun_force) | (distance_matrix > upper_bound)] = 0` is also an option. Yours is more readable. Do you think it's worth profiling?

Reply:
It would work indeed. I'll let optimization for latter, though. We need a general profiling pass anyway.